### PR TITLE
CAZB-1159 Create a proxy for the whitelisting service

### PR DESCRIPTION
**Pull request by radekmorytko**

### CODE REVIEW

File: src/main/java/uk/gov/caz/psr/configuration/AsyncRestConfiguration.java [hunk 1]
```diff
@@ -58,12 +55,33 @@ public AccountsRepository accountsRepository(ObjectMapper objectMapper,
...
+  }
+
+  /**
+   * Creates {@link Retrofit} based on the passed params with the Jackson converter.
+   */
+  private Retrofit buildRetrofit(ObjectMapper objectMapper, String whitelistRootUrl,
+      int readTimeoutSeconds, int connectTimeoutSeconds) {
     return new Retrofit.Builder()
-        .baseUrl(requireNonNull(HttpUrl.parse(formatUrl(accountsApiEndpoint))))
+        .baseUrl(requireNonNull(HttpUrl.parse(formatUrl(whitelistRootUrl))))
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/configuration/AsyncRestConfiguration.java:81]: This is more generic url, not only whitelist

File: src/main/java/uk/gov/caz/psr/dto/whitelist/WhitelistedVehicleResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+import lombok.Value;
+
+/**
+ * Response from the whitelist service for a 'GET v1/whitelisting/vehicles/{vrn}' call.
+ */
+@Value
+public class WhitelistedVehicleResponseDto {
+  String vrn;
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/dto/whitelist/WhitelistedVehicleResponseDto.java:13]: Javadocs

File: src/main/java/uk/gov/caz/psr/repository/WhitelistRepository.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+   * Gets details of a whitelisted vehicle by its {@code vrn} in an asynchronous manner.
+   */
+  @Headers("Accept: application/json")
+  @GET("v1/whitelisting/vehicles/{vrn}")
+  Call<WhitelistedVehicleResponseDto> getWhitelistVehicleDetails(@Path("vrn") String vrn);
+
+  /**
+   * Gets details of a whitelisted vehicle by its {@code vrn} in a synchronous manner.
+   */
+  default Response<WhitelistedVehicleResponseDto> getWhitelistVehicleDetailsSync(String vrn) {
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/repository/WhitelistRepository.java:27]: Can you please use [CacheableResponse](https://github.com/InformedSolutions/JAQU-CAZ-Java-Common/blob/develop/definitions/src/main/java/uk/gov/caz/definitions/dto/CacheableResponseDto.java) ?

File: src/main/java/uk/gov/caz/psr/configuration/AsyncRestConfiguration.java (hunk 1 above)
Comment (radekmorytko) [reply src/main/java/uk/gov/caz/psr/configuration/AsyncRestConfiguration.java:81]: Good catch, fixed

File: src/main/java/uk/gov/caz/psr/repository/WhitelistRepository.java (hunk 3 above)
Comment (radekmorytko) [reply src/main/java/uk/gov/caz/psr/repository/WhitelistRepository.java:27]: Done

**Pull request merged**